build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1185,6 +1371,140 @@ message ExecuteRequest {
+  // to the CAS.
+  message DynamicFileContentRequest {
+    // The file path for which the worker does not have the contents.
+    string path = 1;
+
+    // The file hash of the content expected by the worker.
+    // If this is not the current hash of the file content at `path`
+    string FileHash = 2;
+  }
+
+  // The current stage of execution.
+  ExecutionStage.Value stage = 1;
+
+  // The digest of the [Action][build.bazel.remote.execution.v2.Action]
+  // being executed.
+  Digest action_digest = 2;
+
+  // A series of incremental (streamed) stdout and stderr outputs in order of
+  // write from the action during execution. The full set of console outputs
+  // from the action will also be written in the final
+  // [ExecuteResponse][build.bazel.remote.execution.v2.ExecuteResponse].
+  // The server MAY buffer these outputs in order to return larger batches
+  // of messages less frequently. The server MAY choose to omit incremental
+  // messages and instead wait to the final `ExecuteResponse` to return all
+  // outputs from the action, however in this case the server MUST NOT return
+  // any incremental messages in this field.
+  repeated ConsoleOutput streamed_console_outputs = 3;
+
+  // A directory path for which the worker does not have metadata. this
+  // indicates a directory that the `Action` was not predicted to use
+  // from the client filesystem. The client is requested to enumerate
+  // directory metadata and respond in
+  // [StreamedDynamicExecutionRequest.dynamic_directory_metadata_response][build.bazel.remote.execution.StreamedDynamicExecutionRequest].
+  // The client MAY include individual file digests but this is not required,
+  // the worker will request individual file hashes as needed using
+  // `dynamic_file_hash_path`.
+  string dynamic_directory_content_path = 4;
+
+  // A file path for which the worker does not have a hash. This indicates
+  // a file that the `Action` was not predicted to use from the client
+  // filesystem. The client is requested to hash the file and respond in
+  // [StreamedDynamicExecutionRequest.dynamic_file_hash_response][build.bazel.remote.execution.StreamedDynamicExecutionRequest].
+  string dynamic_file_hash_path = 5;
+
+  // A file path and expected content hash that the worker does not have in
+  // the CAS. This indicates file content that the `Action` was not predicted
+  // to use from the client, or content that was subsequently removed from
+  // the CAS. The client is requested to upload the file content into the
+  // CAS and respond in 
+  // [StreamedDynamicExecutionRequest.dynamic_file_content_response][build.bazel.remote.execution.StreamedDynamicExecutionRequest].
+  DynamicFileContentRequest dynamic_file_content_request = 6;
+}
+
+// A streaming request message for
+// [Execution.ExecuteWithDynamicRetrieval][build.bazel.remote.execution.v2.Execution.ExecuteWithDynamicRetrieval].
+message StreamedDynamicExecutionRequest {
```

Comment:
>StreamedDynamicExecutionRequest [](start = 8, length = 31)

Since this message contains both "Request" and "Response", should it be renamed to `StreamedDynamicExecutionData`?

Reply:
Good idea. Updated.

---
In reply to: [459012092](https://github.com/erikma/remote-apis/pull/1#discussion_r459012092) [](ancestors = 459012092)